draft-ietf-tls-tls13.md

```diff
@@ -3546,35 +3546,32 @@ that remains is to calculate the key schedule.
 
 The TLS handshake establishes one or more input secrets which
 are combined to create the actual working keying material, as detailed
-below. The key derivation process makes use of the following functions,
-based on HKDF {{RFC5869}}:
+below.
+The key derivation process makes use of the HKDF-Extract and HKDF-Expand
+functions as defined for HKDF {{RFC5869}}, as well as the functions
+defined below:
 
 ~~~~
-  HKDF-Extract(Salt, IKM) as defined in {{RFC5869}}.
-  
-  HKDF-Expand-Label(Secret, Label, Messages, Length) =
+  HKDF-Expand-Label(Secret, Label, HashValue, Length) =
```

Comment:
On Monday, May 23, 2016 05:55:47 pm ekr wrote:
> This was meant to be (generally) Hash(Messages) + Hash(resumption_context).

This is the result of this change, as the 3rd arg passed here below in Derive-Secret is "Hash(Messages) + Hash(resumption_context)". Leaving this as it was and changing hash_value to "Hash(Messages)" results in "Hash(Hash(Messages) + Hash(resumption_context))" (hash of concat of hashes). All other uses of HKDF-Expand-Label provide either an empty string for the 3rd arg or in the case of Exporters, context_value.